src/network/server.rs

```diff
@@ -415,10 +415,17 @@ impl MyHubService {
         }
     }
 
-    fn rewrite_hub_event(mut hub_event: HubEvent, shard_index: u32) -> HubEvent {
+    fn rewrite_hub_event(
+        mut hub_event: HubEvent,
+        shard_index: u32,
+        timestamp: Option<u64>,
+    ) -> HubEvent {
         let (block_number, _) = HubEventIdGenerator::extract_height_and_seq(hub_event.id);
         hub_event.block_number = block_number;
         hub_event.shard_index = shard_index;
+        if let Some(timestamp) = timestamp {
+            hub_event.timestamp = timestamp;
+        }
```

Comment:
We can delete this logic 3 days after releasing this right? Worth adding a todo.

Reply:
This logic is because the events we emit live already have timestamp populated

Reply:
We don't write timestamp to disk right now